File: android/src/main/kotlin/com/pravera/flutter_foreground_task/models/ForegroundTaskEventType.kt
package com.pravera.flutter_foreground_task.models

enum class ForegroundTaskEventType(val value: Int) {
    NOTHING(1),
    ONCE(2),
    REPEAT(3);

    companion object {
        fun fromValue(value: Int) =
            ForegroundTaskEventType.values().firstOrNull { it.value == value } ?: NOTHING
    }
}

File: ios/flutter_foreground_task/Sources/flutter_foreground_task/models/ForegroundTaskEventType.swift
//
//  ForegroundTaskEventType.swift
//  flutter_foreground_task
//
//  Created by Woo Jin Hwang on 8/28/24.
//

import Foundation

enum ForegroundTaskEventType: Int {
  case NOTHING = 1
  case ONCE = 2
  case REPEAT = 3
  
  static func fromValue(_ value: Int) -> ForegroundTaskEventType {
    return ForegroundTaskEventType(rawValue: value) ?? .NOTHING
  }
}
